lib/models/asset-size-printer.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+var chalk    = require('chalk');
+var Task     = require('../models/task');
+var fs       = require('fs');
+var path     = require('path');
+var filesize = require('filesize');
+var glob     = require('glob');
+var Promise  = require('../ext/promise');
+
+module.exports = Task.extend({
+  files: [],
+
+  init: function () {
+    this.originalOutputPath = path.normalize(this.outputPath);
+    this.outputPath = path.join(this.outputPath, 'assets');
+  },
+
+  print: function () {
+    var ui = this.ui;
+
+    return Promise.resolve()
+      .then(this.makeFileGlob.bind(this))
+      .then(this.validateAssetPath.bind(this))
+      .then(function () {
+        ui.writeLine(chalk.green('File sizes:'));
+
+        this.files
+          // Skip test files
+          .filter(function (file) {
+            var filename = path.basename(file);
+            return !/test-(loader|support)/i.test(filename);
+          })
+          // Print human-readable file sizes
+          .forEach(function (file) {
+            var filename = path.basename(file);
+            var stat = fs.statSync(file);
+
+            ui.writeLine(chalk.blue(filename + ': ') + chalk.white(filesize(stat.size)));
+          });
+      }.bind(this))
+      .catch(function (err) {
+        ui.writeLine(chalk.red(err));
```

Comment:
typically we want the default error handler in ember-cli to handle this. So that: https://github.com/ember-cli/ember-cli/blob/master/lib/cli/cli.js#L150-L156 can print the errors nicely, and ensure the program exists with the right status code.

I believe you should just be able to omit the `catch` here.